openedx/openedx-learning
# Tagging: serialize object permissions to REST API [FC-0036]

## Description

Adds "user permissions" to the Tagging REST API to show what permissions the current request.user has on the objects serialized in the response.

## Supporting information

Part of https://github.com/openedx/modular-learning/issues/160

## Testing instructions

See https://github.com/openedx/frontend-app-course-authoring/pull/787 for manual testing instructions.

## Deadline

None

## Other information

TO DO (Re [comment](https://github.com/openedx/openedx-learning/pull/138#issuecomment-1879332200):)

- [x] _Please include a test that loads a bunch of taxonomies/tags/objecttags as a non-superuser and makes sure the query count is only increased by at most 1 when permissions are included in the response._

   Done, see [test_views.py](https://github.com/openedx/openedx-learning/pull/138/files#diff-2f11bfdf39bf0bee3b4778c0d84478fa6afd170e57245f804c77b2e9b4d865a4).

- [x] _Do we want to allow API consumers to choose whether the perms info is included in the REST API responses or not?
   It's probably simpler to just always include them, unless there is some performance or response size penalty._
   
   I decided to add an `include_perms` parameter which triggers the permission checks. These checks are cheap to run in this library, but users of the library may customize their permissions (like we did with content_tagging) and so the checks may not always be this quick to run.
   

Private ref: FAL-3577
